src/main/java/pt/tqs/g205/services/CategoriaPratoService.java

```diff
@@ -0,0 +1,46 @@
+package pt.tqs.g205.services;
+
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.stereotype.Service;
+
+import pt.tqs.g205.domain.CategoriaPrato;
+import pt.tqs.g205.repositories.CategoriaPratoRepository;
+
+import java.util.List;
+import java.util.NoSuchElementException;
+import java.util.Optional;
+
+/**
+ * Servicos para a classe CategoriaPrato.
+ */
+@Service
+public class CategoriaPratoService {
+
+  @Autowired
+  private CategoriaPratoRepository categoriaRepo;
+  
+  /**
+   * Obter todas as categorias de pratos.
+   * @return lista de categorias.
+   */
+  public List<CategoriaPrato> getAll() {
+    List<CategoriaPrato> categorias = categoriaRepo.findAll();
+    categorias.forEach(e -> e.setPratos(null));
+    return categorias;
+  }
+  
+  /**
+   * Obter uma categoria pelo seu id.
+   * @param id id da categoria.
+   * @return categoria.
+   */
+  public CategoriaPrato getById(Integer id) throws NoSuchElementException {
+    Optional<CategoriaPrato> cat = categoriaRepo.findById(id);
+
+    if(cat.get() == null)
```

Comment:
![Bug](https://sonarsource.github.io/sonar-github/bug.png 'Bug') Bug: Call "cat.isPresent()" before accessing the value. ([squid:S3655](https://sonarcloud.io/organizations/bdomingues-github/rules#rule_key=squid%3AS3655))

[See it in SonarCloud](https://sonarcloud.io/project/issues?id=pt.tqs.g205%3Arestapi&pullRequest=5&issues=AWNDDljGXOBV_Yrh-1IV&open=AWNDDljGXOBV_Yrh-1IV)

[ISSUE_KEY]: AWNDDljGXOBV_Yrh-1IV